### PR TITLE
Stabilize AgentSet

### PR DESCRIPTION
**Pull request by EwoutH**
Removes the experimental warning for the AgentSet.

### CODE REVIEW

File: mesa/model.py [hunk 1]
```diff
@@ -76,9 +76,6 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._steps: int = 0
         self._time: TimeT = 0  # the model's clock
 
-        # Warning flags for current experimental features. These make sure a warning is only printed once per model.
-        self.agentset_experimental_warning_given = False
-
```
Comment (quaquel) [inline mesa/model.py:81]: are you sure this flag is not used for some other warnings?

File: mesa/agent.py [hunk 2]
```diff
@@ -115,18 +108,11 @@ def __init__(self, agents: Iterable[Agent], model: Model):
             agents (Iterable[Agent]): An iterable of Agent objects to be included in the set.
             model (Model): The ABM model instance to which this AgentSet belongs.
         """
-        self.model = model
+        self._agents = weakref.WeakKeyDictionary()
+        for agent in agents:
+            self._agents[agent] = None
```
Comment (quaquel) [inline mesa/agent.py:113]: why not do this loop as a dict expression as previously?

Comment (EwoutH) [reply mesa/agent.py]: Not intended. Got a merge conflict, solved it the wrong way. Should be fixed now.

**Review by quaquel — APPROVED** [review]

**Comment by Corvince** [comment]
Thanks!

**Pull request merged**